pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -906,7 +910,17 @@ func (v *Value) MarshalJSON() ([]byte, error) {
 	case v.Ap != nil:
 		return json.Marshal(v.Ap)
 	case v.List != nil:
-		return json.Marshal(v.List)
+		// Buildexpression list order does not matter, so sorting is necessary for
```

Comment:
This is the kicker. When marshaling buildexpression's to JSON, lists need to be sorted since order doesn't matter. That way we can perform comparisons for equality.